raiden_installer/token_exchange.py

```diff
@@ -106,83 +175,47 @@ def get_current_rate(self, token_amount: TokenAmount) -> EthereumAmount:
 
         return EthereumAmount(Wei(max(expected_rate, slippage_rate)))
 
-    def _calculate_transaction_costs(self, token_amount: TokenAmount, account: Account) -> dict:
-        log.debug("calculating exchange rate")
-        exchange_rate = self.get_current_rate(token_amount)
-
-        eth_sold = EthereumAmount(token_amount.value * exchange_rate.value * Decimal(1.2))
-        log.debug("calculating gas price")
+    def _get_gas_price(self):
         web3_gas_price = self.w3.eth.generateGasPrice()
         kyber_max_gas_price = self.network_contract_proxy.functions.maxGasPrice().call()
         max_gas_price = min(web3_gas_price, kyber_max_gas_price)
-        gas_price = EthereumAmount(Wei(max_gas_price))
-        token_network_address = self.get_token_network_address(token_amount.ticker)
-        transaction_params = {
-            "from": account.address,
-            "value": eth_sold.as_wei,
-            "gasPrice": gas_price.as_wei,
-        }
+        return EthereumAmount(Wei(max_gas_price))
+
+    def _estimate_gas(
+        self,
+        token_amount: TokenAmount,
+        account: Account,
+        transaction_params: dict,
+        **kw
+    ):
+        exchange_rate = kw["exchange_rate"]
```

Comment:
This could lead to an key error. Can you handle the error?
```suggestion
        exchange_rate = kw.get("exchange_rate", None)
        if exchange_rate is None:
            # handle error
```
or

```suggestion
        if "exchange_rate in kw:
             exchange_rate = kw["exchange_rate"]
```
or
```suggestion
        try:
            exchange_rate = kw["exchange_rate"]
        except KeyError:
            #handle 
```

Reply:
This was intentional as it is just a private method. Would you rather throw a more explicit `ExchangeError` in that case?

Reply:
also possible, even better.